lib/lbt/resources/ResourceCollector.js

```diff
@@ -279,37 +279,43 @@ class ResourceCollector {
 
 		await Promise.all(promises);
 
-		const debugBundlePromises = [];
-
-		for (let i = debugResourcesInfo.length - 1; i >= 0; i--) {
-			const dbgInfo = debugResourcesInfo[i];
-			const nonDebugName = ResourceInfoList.getNonDebugName(dbgInfo.name);
+		await Promise.all(debugResourcesInfo.map(async (dbgInfo) => {
+			const debugName = dbgInfo.name;
+			const nonDebugName = ResourceInfoList.getNonDebugName(debugName);
 			const nonDbgInfo = this._resources.get(nonDebugName);
 
 			// FIXME: "merged" property is only calculated in ResourceInfo#copyFrom
 			// Therefore using the same logic here to compute it.
 			if (!nonDbgInfo || (nonDbgInfo.included != null && nonDbgInfo.included.size > 0)) {
 				// We need to analyze the dbg resource if there is no non-dbg variant or
 				// it is a bundle because we will (usually) have different content.
-				debugBundlePromises.push(
-					this.enrichWithDependencyInfo(dbgInfo)
-				);
+
+				// In order to retrieve the correct raw module info, the name must be set
+				// to the non-debug name. The same happens during bundling, where also only
+				// the non-debug names are used.
+				dbgInfo.name = nonDebugName;
```

Comment:
Changing the name here causes the ResourcePool to return the cached non-debug info instead of using that name to analyze the debug file.

So we need another solution here. Probably by checking for the IsDebugVariant tag to decide whether to analyze the resource.

Reply:
This should be fixed now